### PR TITLE
always return a Buffer for readFile

### LINKED ISSUE #92: readFile() should always return a Buffer
We shouldn't try to guess the encoding of a file. For consistency, we should always return a Buffer, letting the user choose how to deal with it.

### LINKED ISSUE #94: readFile result for binary data
May be related to #92 : I do a unifile.readFile then fs.writeFile and the new image is changed... Or it may be a problem of encoding?

I do something like this:
```
unifile.readFile(session, 'fs', 'preview1.jpeg').then(result => {
    fs.writeFileSync('/home/lexoyo/preview2.jpeg', result, 'binary');
});
```
And the images differ:
preview1.jpeg starts with `ÿØÿà^@^PJFIF^@^A^A^A^@H^@H^@^@ÿÛ^@C^@^C^B^B^C^B^B^C^C^C^C^D^C^C^D^E^H^E^E^D^D^E` whereas preview2.jpeg starts with `ýýýý^@^PJFIF^@^A^A^A^@H^@H^@^@ýý^@C^@^C^B^B^C^B^B^C^C^C^C^D^C^C^D^E^H^E^E^D^D^E`

### PR DESCRIPTION
**Pull request by lexoyo**
Closes #92 
Closes #94

### CODE REVIEW

**Comment by lexoyo** [comment]
based on the branch `wip-coverage` in order to pass unit tests

**Comment by lexoyo** [comment]
@JbIPS the bitHound service says `Method 'batch' has a complexity of 12. ` and `Function has a complexity of 9. `

And codacity says `Function has a complexity of 9. (complexity)` 

But I did not touch these parts of the code, did I?

**Comment by lexoyo** [comment]
also I did not test the webdav and remoteStorage and sftp services but applied the same changes than those to dropbox, github, fs and ftp

File: lib/unifile-sftp.js [hunk 1]
```diff
@@ -199,7 +199,7 @@ class SftpConnector {
 		const sftp = sftpSession ? new SFTPClient() : new SFTPClient(session);
 		return sftp.getBuffer(path, sftpSession)
 		.catch(parseError)
-		.then((buffer) => buffer.toString());
+		.then((buffer) => buffer);
```
Comment (JbIPS) [inline lib/unifile-sftp.js]: This is useless, just return.

File: test/unifile-fs.js [hunk 2]
```diff
@@ -367,7 +367,7 @@ describe('FsConnector', function() {
 		it('returns the content of a file', function() {
 			return connector.readFile({}, '/tmp/test')
 			.then((content) => {
-				expect(content).to.equal(data);
+				expect(content.toString()).to.equal(data);
```
Comment (JbIPS) [inline test/unifile-fs.js]: Could you add `expect(content).to.be.an.instanceof(Buffer)` please?

File: test/unifile-ftp.js [hunk 3]
```diff
@@ -387,7 +387,10 @@ describe('FtpConnector', function() {
 		});
 
 		it('returns the content of a file', function() {
-			return connector.readFile(session, 'tmp.test').should.become(data);
+			return connector.readFile(session, 'tmp.test')
+			.then((content) => {
+				expect(content.toString()).to.equal(data);
```
Comment (JbIPS) [inline test/unifile-ftp.js]: Same here (and everywhere)

**Comment by JbIPS** [comment]
Don't worry about complexity, I think that's from my work. You still have some issues I pointed.

Thank you for this much needed fix! 🙇

File: lib/unifile-sftp.js (hunk 1 above)
Comment (lexoyo) [reply lib/unifile-sftp.js]: :D

File: test/unifile-fs.js (hunk 2 above)
Comment (lexoyo) [reply test/unifile-fs.js]: sure

**Comment by lexoyo** [comment]
Done :)

**Comment by lexoyo** [comment]
Thx for taking the time to merge

**Pull request merged**